sdk/ai/azure-ai-projects/samples/agents/async_samples/sample_agents_azure_functions_async.py

```diff
@@ -0,0 +1,110 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import asyncio
+
+"""
+FILE: sample_agents_azure_functions_async.py
+ 
+DESCRIPTION:
+    This sample demonstrates how to use azure function agent operations from
+    the Azure Agents service using a asynchronous client.
+ 
+USAGE:
+    python sample_agents_azure_functions_async.py
+ 
+    Before running the sample:
+ 
+    pip install azure-ai-projects azure-identity
+ 
+    Set this environment variables with your own values:
+    PROJECT_CONNECTION_STRING - the Azure AI Project connection string, as found in your AI Studio Project.
+    STORAGE_SERVICE_ENDPONT - the storage service queue endpoint, triggering Azure function.
+    Please see Getting Started with Azure Functions page for more information on Azure Functions:
+    https://learn.microsoft.com/azure/azure-functions/functions-get-started
+"""
+
+import os
+from azure.ai.projects.aio import AIProjectClient
+from azure.identity.aio import DefaultAzureCredential
+from azure.ai.projects.models import (
+    AzureFunctionStorageQueue,
+    AzureFunctionTool,
+)
+
+
+async def main():
+
+    async with DefaultAzureCredential(
+        exclude_managed_identity_credential=True, exclude_environment_credential=True
+    ) as creds:
+        async with AIProjectClient.from_connection_string(
+            credential=creds,
+            conn_str=os.environ["PROJECT_CONNECTION_STRING"],
+        ) as project_client:
+
+            storage_service_endpoint = os.environ["STORAGE_SERVICE_ENDPONT"]
+            azure_function_tool = AzureFunctionTool(
+                name="foo",
+                description="Get answers from the foo bot.",
+                parameters={
+                    "type": "object",
+                    "properties": {
+                        "query": {"type": "string", "description": "The question to ask."},
+                        "outputqueueuri": {"type": "string", "description": "The full output queue uri."},
+                    },
+                },
+                input_queue=AzureFunctionStorageQueue(
+                    queue_name="azure-function-foo-input",
+                    storage_service_endpoint=storage_service_endpoint,
+                ),
+                output_queue=AzureFunctionStorageQueue(
+                    queue_name="azure-function-tool-output",
+                    storage_service_endpoint=storage_service_endpoint,
+                ),
+            )
+
+            agent = await project_client.agents.create_agent(
+                model="gpt-4",
```

Comment:
We should load model name from env variables, as we do in inference/connection samples. But this can be done after the release.

Reply:
Created task for that: https://github.com/orgs/ai-platform-ml-platform/projects/7/views/2?sliceBy%5Bvalue%5D=nirovins_microsoft&pane=issue&itemId=90741293